var/spack/repos/builtin/packages/regcm/package.py

```diff
@@ -23,17 +23,24 @@ class Regcm(AutotoolsPackage):
 
     # On Intel and PGI compilers, multiple archs can be built at the same time,
     # producing a so-called fat binary. Unfortunately, gcc builds only the last
-    # architecture provided (in the configure), so we allow a single arch.
+    # architecture provided (in the configure), so we allow a single arch on
+    # GCC. Only GCC and Intel are supported.
```

Comment:
I don't really understand the comment. At the beginning it seems PGI compilers are allowed. At the end it seems that only GCC and Intel can compile this package.

Reply:
See this discussion with @adamjstewart: https://github.com/spack/spack/pull/9934#pullrequestreview-178034960
and also this #974 (in particular this comment: https://github.com/spack/spack/issues/974#issuecomment-225247436)

Basically, Intel and PGI can build multi-arch binaries, while GCC cannot.
Moreover, this package (RegCM) has optimizations defined only for Intel and GCC, but should be buildable also with other compilers.

To sum up:
- with intel, a user should be able to build a single executables for all the combinations of architectures they want (using `--extension=knl,skl,bdw,nhl`);
- with gcc, a user is allowed to build an executable using a single optimization;
- with any other compiler, no _extensions/optimizations_ are supported.

That's also the reason for those "strange" checks in `configure_args`.

I can rephrase that comment better.

Reply:
Hope the comment in commit 75fdc14f clarifies.